.eslintrc.cjs

```diff
@@ -31,6 +41,33 @@ module.exports = {
         argsIgnorePattern: '^_',
         varsIgnorePattern: '^_'
       }
-    ]
+    ],
+    '@typescript-eslint/restrict-template-expressions': [
+      'error',
+      {
+        allowNever: true
+      }
+    ],
+
+    // Todo: investigate whether we'd like these on
+    '@typescript-eslint/ban-ts-comment': 'off',
+    '@typescript-eslint/ban-types': 'off',
+    '@typescript-eslint/consistent-indexed-object-style': 'off',
+    '@typescript-eslint/consistent-type-definitions': 'off',
+    '@typescript-eslint/no-confusing-void-expression': 'off',
+    '@typescript-eslint/no-empty-function': 'off',
+    '@typescript-eslint/no-floating-promises': 'off',
+    '@typescript-eslint/no-unsafe-argument': 'off',
+    '@typescript-eslint/no-unsafe-assignment': 'off',
+    '@typescript-eslint/no-unsafe-call': 'off',
+    '@typescript-eslint/no-unsafe-member-access': 'off',
+    '@typescript-eslint/no-unsafe-return': 'off',
+    '@typescript-eslint/prefer-function-type': 'off',
+    '@typescript-eslint/sort-type-constituents': 'off',
+    '@typescript-eslint/unbound-method': 'off',
+    'prefer-rest-params': 'off',
```

Comment:
Each of these had at least a few existing warnings, so I didn't want to overreach and solve for them without asking. Are any of these rules ones you feel particularly strongly about?

* https://typescript-eslint.io/rules/ban-ts-comment
* https://typescript-eslint.io/rules/ban-types
* https://typescript-eslint.io/rules/consistent-indexed-object-style
* https://typescript-eslint.io/rules/consistent-type-definitions
* https://typescript-eslint.io/rules/no-confusing-void-expression
* https://typescript-eslint.io/rules/no-empty-function
* https://typescript-eslint.io/rules/no-floating-promises
* https://typescript-eslint.io/rules/no-unsafe-argument
* https://typescript-eslint.io/rules/no-unsafe-assignment
* https://typescript-eslint.io/rules/no-unsafe-call
* https://typescript-eslint.io/rules/no-unsafe-member-access
* https://typescript-eslint.io/rules/no-unsafe-return
* https://typescript-eslint.io/rules/prefer-function-type
* https://typescript-eslint.io/rules/sort-type-constituents
* https://typescript-eslint.io/rules/unbound-method
* https://eslint.org/docs/latest/rules/prefer-rest-params

If not, I can go ahead and fix for all their complaints.